docs/tutorials/onnx/inference_on_onnx_model.md

```diff
@@ -0,0 +1,259 @@
+    Input format: (1, 3, 224, 224) <!--no-notebook-->
+
+
+We can visualize the network (requires graphviz installed)
+
+
+```python
+mx.visualization.plot_network(sym, shape={"input_0":inputs[0].shape}, node_attrs={"shape":"oval","fixedsize":"false"})
+```
+
+
+
+
+![png](https://github.com/ThomasDelteil/web-data/blob/c77c2e93ba142f45682ed63c191d2568b20aff25/mxnet/doc/tutorials/onnx/network.png?raw=true)<!--no-notebook-->
+
+
+
+This is a helper function to run a batch of data through the net and collate the outputs
+
+
+```python
+def run_batch(net, data):
+    results = []
+    for batch in data:
+        outputs = net(batch)
+        for output in outputs.asnumpy(): results.append(output)
+    return np.array(results)
+```
+
+
+```python
+result = run_batch(net, nd.array(inputs, ctx))
+```
+
+
+```python
+print("Loaded model and sample output predict the same class: {}".format(np.argmax(result) == np.argmax(outputs[0])))
+```
+
+    Loaded model and sample output predict the same class: True <!--no-notebook-->
+
+
+Good the sample output and our prediction match, now we can run against real data
+
+## Test using real images
+
+
+```python
+TOP_N = 3 # How many top guesses we show in the visualization
+```
+
+
+Transform function to set the data into the format the network requires (N, 3, 224, 224)
+
+
+```python
+transform = lambda img: np.expand_dims(np.transpose(img, (2,0,1)),axis=0).astype(np.float32)
+```
+
+
+We load two sets of images in memory
+
+
+```python
+image_net_images = [plt.imread('images/{}.jpg'.format(path)) for path in ['apron', 'hammerheadshark','dog']]
+caltech101_images = [plt.imread('images/{}.jpg'.format(path)) for path in ['wrench', 'dolphin','lotus']]
+images = image_net_images + caltech101_images
+```
+
+And run them as a batch through the network to get the predictions
+
+```python
+batch = nd.array(np.concatenate([transform(img) for img in images], axis=0), ctx=ctx)
+result = run_batch(net, [batch])
+```
+
+
+```python
+plot_predictions(image_net_images, result[:3], categories, TOP_N)
+```
+
+
+![png](https://github.com/ThomasDelteil/web-data/blob/c77c2e93ba142f45682ed63c191d2568b20aff25/mxnet/doc/tutorials/onnx/imagenet.png?raw=true)<!--no-notebook-->
+
+
+**Well done!** Looks like it is doing a pretty good job at classifying pictures when the category is a ImageNet label
+
+Let's now see the results on the 3 other images
+
+
+```python
+plot_predictions(caltech101_images, result[3:7], categories, TOP_N)
+```
+
+
+![png](https://github.com/ThomasDelteil/web-data/blob/c77c2e93ba142f45682ed63c191d2568b20aff25/mxnet/doc/tutorials/onnx/caltech101.png?raw=true)<!--no-notebook-->
+
+
+**Hmm, not so good...** We see where the network is coming from but effectively `wrench`, `dolphin` and `lotus` categories are not in the ImageNet classes...
+
+Lucky for us, the [Caltech101 dataset](http://www.vision.caltech.edu/Image_Datasets/Caltech101/) has them, let's see if we can fine-tune our network to classify those correctly.
```

Comment:
let's see **how** we can...classify **these categories** correctly.

Reply:
:+1: